mlflow/transformers.py

```diff
@@ -426,7 +428,10 @@ def save_model(
         flavor_conf.update({_PROCESSOR_TYPE_KEY: _get_instance_type(processor)})
 
     # Save the pipeline object
-    built_pipeline.save_pretrained(save_directory=path.joinpath(_PIPELINE_BINARY_FILE_NAME))
+    built_pipeline.model.save_pretrained(
+        save_directory=path.joinpath(_MODEL_BINARY_FILE_NAME),
+        max_shard_size=MLFLOW_HUGGINGFACE_MODEL_MAX_SHARD_SIZE.get(),
```

Comment:
Can we also check if the parameter is passed in in `kwargs`, so that we can support folks overriding it?

Reply:
They can override it with the env var now. Do we want to expose it in `mlflow.transformers.save_model` too? I think it depends on how easy we want to allow users to override this, and I thought we don't want it to be too easy. 
Let me know if we want 1) env var 2) kwargs or 3) both. I can change accordingly.

Reply:
I think restricting this behavior to purely environment variables is fine. Can we add a note in `models.rst` explaining the behavior and the env var key that would need to be overridden for edge cases to change the sharding size?